src/linker/Linker.Steps/DiscoverCustomOperatorsHandler.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class DiscoverOperatorsHandler : IMarkHandler
+	{
+		LinkContext _context;
+		bool markOperators;
+		HashSet<TypeDefinition> _trackedTypesWithOperators;
+		Dictionary<TypeDefinition, HashSet<MethodDefinition>> _pendingOperatorsForType;
+
+		public void Initialize (LinkContext context, MarkContext markContext)
+		{
+			_context = context;
+			_trackedTypesWithOperators = new HashSet<TypeDefinition> ();
+			_pendingOperatorsForType = new Dictionary<TypeDefinition, HashSet<MethodDefinition>> ();
+			markContext.RegisterMarkTypeAction (ProcessType);
+		}
+
+		void ProcessType (TypeDefinition type)
+		{
+			CheckForLinqExpressions (type);
```

Comment:
Shouldn't this return if the type is not SLE?

Reply:
It shouldn't return but it can skip some logic - good catch. Fixed.